test/appium/views/transactions_view.py

```diff
@@ -18,23 +18,30 @@ def __init__(self, driver):
             super(OptionsButton.OpenOnEtherscanButton, self).__init__(driver)
             self.locator = self.Locator.xpath_selector("//*[@text='Open on Etherscan.io']")
 
+
 class TransactionTable(BaseElement):
     def __init__(self, driver):
         super(TransactionTable, self).__init__(driver)
         self.driver = driver
         self.locator = self.Locator.xpath_selector("//android.support.v4.view.ViewPager")
 
     class TransactionElement(BaseButton):
-        def __init__(self, driver, selector):
+        def __init__(self, driver):
             super(TransactionTable.TransactionElement, self).__init__(driver)
-            self.driver = driver
-            if 'amount' in selector:
-                self.locator = self.Locator.xpath_selector(
-                    "(//android.widget.TextView[contains(@text,'%s ETH')])" % selector['amount'])
-            elif 'index' in selector:
-                self.locator = self.Locator.xpath_selector(
-                    '(//android.view.ViewGroup[@content-desc="transaction-item"])[%d]' % (selector['index'] + 1))
 
+        @staticmethod
+        def by_amount(driver, amount: str):
```

Comment:
@antdanchenko I think using [factory methods](https://pythonspot.com/factory-method/) is a nice solution here. Otherwise, if I'd left `amount` selector in `TransactionElement` constructor I'd be then overriding it in `get_first_transaction`. Having factories for `TransactionElement` is much cleaner.